### DISCUSSION TITLE
Global imports

### DISCUSSION THREAD

**Discussion by Davidster** [Davidster/ikari#23] [General]
Currently all the modules are automatically imported globally. I'd like to remove all or most of the `use super::*` imports to make the module dependencies more clear and when reading the code you won't have to jump to definition to see which module the code comes from.

What do you think @CatCode79 ?

**Comment by CatCode79** [comment]
I can't indicate what possible disadvantages we would have, I've never used super (for the same reason you opened this thread).

From this example super is used to handle ambiguity, but just give different module names and we shouldn't run into problems:
https://doc.rust-lang.org/rust-by-example/mod/super.html

I think we can do without it, it probably makes more sense to use it in complex module trees or if one doesn't want to depend on module names in a hardcore way: it's easier to refactor however the IDE's refactoring features are so advanced that I don't think is reason enough.

**Reply by Davidster** [reply]
I agree, it's a low-risk change since it's easy to refactor rust code :)
